modules/client/src/main/java/org/apache/ignite/internal/client/table/ClientTable.java

```diff
@@ -762,16 +796,24 @@ private static <E> void reduceWithKeepOrder(List<E> agg, List<E> cur, List<Integ
         }
     }
 
-    <R, E> CompletableFuture<R> split(
-            Transaction tx,
+    <R, E> CompletableFuture<R> splitAndRun(
             Collection<E> keys,
-            BiFunction<Collection<E>, PartitionAwarenessProvider, CompletableFuture<R>> fun,
+            MapFunction<E, R> fun,
             @Nullable R initialValue,
             Reducer<R> reducer,
             BiFunction<ClientSchema, E, Integer> hashFunc
     ) {
-        assert tx != null;
+        return splitAndRun(keys, fun, initialValue, reducer, hashFunc, List.of());
```

Comment:
Should we skip the splitting if there is only one (or too few) active client connections? Looks like splitting might reduce performance in that case.